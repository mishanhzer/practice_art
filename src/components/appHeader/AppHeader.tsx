import React from "react";
import "./appHeader.css";
import "./appHeader.scss";
import { images } from "../../assets/logo/logo";
import { mainLogo, basket, arrow } from "../../assets/logo/logo";

export const AppHeader = () => {
  return (
    <header className={`flex justify-between pt-8 items-center h-28`}>
      <div className={`flex flex-row w-1/4 justify-between`}>
        <div className={`flex w-28 justify-between`}>
          {images.map((item) => {
            return <Img key={item.name} src={item.src} />;
          })}
        </div>
      </div>
      <div>
        <img className={`w-24 rounded-full border-2 border-black border-solid`} src={mainLogo} alt="#" />
      </div>
      <div className={`flex justify-end items-center w-1/4 `}>
        <a className={`w-10`} href='#'>
          <img className={`w-5`} src={basket} alt="#" />
        </a>
          <a className={`flex ml-8 py-6 px-10
             bg-blue-900 
             text-base
              text-white 
              items-center 
              font-serif 
              hover:transition
               hover:bg-blue-500 
              group`} href='#'
            >
              CONTACT ME
              <img className={`ml-2 w-6 h-6 group:hover:translate-x-3`} src={arrow} alt="#" />
          </a>
      </div>
    </header>
  );
};

// export const AppHeader = () => {
//   return (
//     <header className="appHeader">
//       <div className="appHeader__wrapper">
//         <div className="appHeader__wrapper_content">
//           {images.map((item) => {
//             return <Img key={item.name} src={item.src} />;
//           })}
//         </div>
//       </div>
//       <div className="appHeader__img">
//         <img className={`w-16`} src={mainLogo} alt="#" />
//       </div>
//       <div className="appHeader__contact">
//         <a className="appHeader__contact_basket" href='#'>
//           <img className={`w-5`} src={basket} alt="#" />
//         </a>
//           <a className="appHeader__contact_contactBlock" href='#'>
//               CONTACT ME
//               <img src={arrow} alt="#" />
//           </a>
//       </div>
//     </header>
//   );
// };

const Img = ({ key, src }) => {
  return (
    <a key={key} href="#">
      <img className={`w-5 rounded-md`} src={src} alt="#" />
    </a>
  );
};
