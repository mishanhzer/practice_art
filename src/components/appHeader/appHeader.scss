.appHeader {
  display: flex;
  justify-content: space-between;
  padding-top: 30px;
  flex-shrink: 0;
  height: 100px;
  align-items: center;
    &__wrapper {
      display: flex;
      flex: 0 1 25%;
      flex-direction: row;
      max-width: 25%;
      justify-content: space-between;
      order: 1;
        &_content {
          display: flex;
          width: 100px;
          justify-content: space-between;
            img {
              border-radius: 5px;
              width: 20px;
            }
        }
    }
    &__img {
      border: 3px solid white;
      border-radius: 50%;
      justify-content: center;
      order: 2;
        img {
          border: 1px solid black;
          width: 80px;
          border-radius: 50%;
          justify-self: center;
      }
    }
    &__contact {
      display: flex;
      justify-content: end;
      align-items: center;
      max-width: 25%;
      flex: 0 1 25%;
      order: 4;
      &_basket {
        width: 40px;
      }
      &_contactBlock {
        margin-left: 2vh;
        padding: 1.5em 2.5em;
        background: #192e2f;
        font-size: 16px;
        display: flex;
        font-family: Lora, serif;
        color: #ffffff;
        fill: #ffffff;
          img {
            margin-left: .6em;
            height: 1.4em;
            width: 1.4em;
          }
          &:hover {
            font-family: Lora, serif;
            color: #ffffff;
            fill: #ffffff;
            background: rgba(25, 46, 47, 0.7);
            transition: opacity .2sease-out;
              img {
                transform: translateX(7px);
                transition-duration: .2s;
              }
          }
      }
    }
}

