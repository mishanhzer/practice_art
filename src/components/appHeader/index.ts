// Файл для сокращения пути import’а компонента. Также можно использовать для единой точки входа экспортируемых материалов из этого каталога.
export {};
