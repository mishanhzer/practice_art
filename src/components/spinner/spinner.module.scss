.spinnerContainer {
  display: flex; 
  justify-content: center;
  align-items: center;
  height: 500px;
}

.spinner {
  width: 100px;
  height: 100px;
}