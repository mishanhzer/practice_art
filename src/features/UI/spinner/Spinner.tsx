import 'bootstrap/dist/css/bootstrap.min.css';

import styles from './spinner.module.scss'

export const Spinner = () => {
  return (
    <div className={styles.container}>
      <div className={`${styles.spinner} spinner-border text-primary`} role="status">
        <span className="sr-only">Loading...</span>
      </div>
    </div>
  )
}
