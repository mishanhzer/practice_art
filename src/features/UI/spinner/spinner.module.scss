.container {
  @apply flex justify-center items-center h-[500px]
}

.spinner {
  @apply w-[100px] h-[100px]
}