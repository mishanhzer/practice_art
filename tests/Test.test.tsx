import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { expect, test } from '@jest/globals';
import { Test } from "./Test";
import { AppHeader } from "../src/components/appHeader/AppHeader";

import { Spinner } from "../src/components/spinner/Spinner";

describe('Spinner component', () => {
  it('Spinner comp', () => {
    const { getByText } = render(<Spinner />)
    expect(getByText('Loading...')).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });

  it('Spinner role', () => {
    const { getByRole } = render(<Spinner />)
    expect(getByRole('status')).toBeInTheDocument()
  })

  it('Spinner correct class', () => {
    const { container } = render(<Spinner />)
    expect(container.firstChild).toHaveClass('flex justify-center items-center h-[500px]')
    expect(container.firstChild?.firstChild).toHaveClass('w-[100px] h-[100px] spinner-border text-primary')
  })
});

// test("render Comp", () => {
//   const { getByText } = render(<Test />);
//   expect(getByText("Hello")).toBeInTheDocument();
// });

// test('Spinner comp', () => {
//   const { getByText } = render(<Spinner />)
//   expect(getByText('Loading...')).toBeInTheDocument()
// })

// describe("AppHeader", () => {
//   test("renders correctly", () => {
//     const { container } = render(<AppHeader />);
//     expect(container).toMatchSnapshot();
//   });
// });

// test("My func", () => {
//   expect([1, 2]).toEqual([1, 2]);
// });
