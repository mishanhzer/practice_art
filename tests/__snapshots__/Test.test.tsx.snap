// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Spinner component renders correctly 1`] = `
<div>
  <div
    class="flex justify-center items-center h-[500px]"
  >
    <div
      class="w-[100px] h-[100px] spinner-border text-primary"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
